packages/fetchai/contracts/erc1155/contract.py

```diff
@@ -77,38 +79,39 @@ def create_token_ids(self, token_type: int, nb_tokens: int) -> List[int]:
         :param nb_tokens: the number of tokens
         :return: the list of token ids newly created
         """
-        lowest_valid_integer = 1
-        token_id = Helpers().generate_id(lowest_valid_integer, token_type)
+        lowest_valid_integer = Helpers().get_next_min_index(self.token_id_to_type)
         token_id_list = []
         for _i in range(nb_tokens):
+            token_id = Helpers().generate_id(lowest_valid_integer, token_type)
             while self.instance.functions.is_token_id_exists(token_id).call():
                 # token_id already taken
                 lowest_valid_integer += 1
                 token_id = Helpers().generate_id(lowest_valid_integer, token_type)
             token_id_list.append(token_id)
-            self.token_ids[token_id] = token_type
+            self._token_id_to_type[token_id] = token_type
             lowest_valid_integer += 1
-            token_id = Helpers().generate_id(lowest_valid_integer, token_type)
         return token_id_list
 
-    def get_deploy_transaction(
+    def get_deploy_transaction_msg(
```

Comment:
We now always have two apis for each operation; one returns a tx object, the other returns a tx msg object